### ISSUE TITLE
Feature Idea: Backup

### ISSUE THREAD

**Issue by ZevSwe** [mattermost/focalboard#253]
**Summary:**
As this is a tool that will work best once one has spent a lot of time and effort setting everything up just as one want a good backup would be nice to have. I know there's an export function but as always that is only as strong as the person behind it's routines. As there are a export function I however can't imagine the feature would be hard to build.

My proposal is that the server can run an export to a local folder on the server every x time. Then that folder can be synced to users backup of choice with whatever backup program he or she wants. This feature could either be built into the GUI or it could just be a small script that one can run through cron. 

**How important this is to me and why:**

Importance: Medium
Work around, take manual backups. Would be nice to get around this in time though. 

Use cases:
1. Backup database to other media in case of emergency.

**Comment by chenilim** [comment]
Thanks for the idea @ZevSwe! This should be fairly easy to do, even with a cron-job or other scripts today, especially with Personal Desktop or when using SQLite. You just need to back up:
* `focalboard.db`
* The `files` folder (for image uploads)

With Postgres, it might be slightly more complex, following its [backup and restore best practices](https://www.postgresql.org/docs/9.1/backup.html). Welcome input from others with more experience with this. What do you do to backup similar systems?

**Comment by ZevSwe** [comment]
I'm not really sure how the export function works. It seems that does some kind of data dump from the database. Could that maybe be used in some way?

**Comment by chenilim** [comment]
@ZevSwe, the export code is defined [here in api.go](https://github.com/mattermost/focalboard/blob/f5ae0b265a8ded60813678938e7705489c7310e4/server/api/api.go#L572). It's pretty much a dump of all the blocks. You can implement a backup solution by calling this API, but it will be missing other things in the database (e.g. users) plus the uploaded images.

**Comment by ZevSwe** [comment]
Right, in that case it seems backing up the database to file would be the only option. It does not look to complicated. Backup the image folder and then the database, found some info here: https://www.tecmint.com/backup-and-restore-postgresql-database/
